### PR TITLE
Implement command line interface

### LINKED ISSUE #108: Command line interface

### PR DESCRIPTION
**Pull request by 0xd34d10cc**
with config integration

Closes #108

### CODE REVIEW

File: src/app/app.cpp [hunk 1]
```diff
@@ -74,20 +62,18 @@ static encoder::Encoder create_encoder(Context context, const sc::Monitor& monit
...
+  const auto host = context.m_config->metrics;
   return std::make_unique<prometheus::Exposer>(host);
 }
 
 static NetworkPtr create_network(Context context) {
-  const auto url_str = context.m_config->get<std::string>("url", "tcp://127.0.0.1:1337");
+  const auto url_str = context.m_config->url;
   const auto url = Url::from_string(url_str);
-
-  context.m_logger.info("Streaming to {}", url.to_string());
```
Comment (pl0q1n) [inline src/app/app.cpp]: Why did you delete this?

File: src/common/options.hpp [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <string>
+#include <cstdlib>
+
+
+namespace shar {
+
+struct Options {
+  std::string url;                           // url to stream to
```
Comment (pl0q1n) [inline src/common/options.hpp]: Should we initialize all options with default values? (url, codec...)

File: conanfile.py [hunk 3]
```diff
@@ -4,14 +4,15 @@
...
-    requires        = ("asio/1.12.0@bincrafters/stable",             # network
+    requires        = ("asio/1.12.0@bincrafters/stable",              # network
                        "ScreenCaptureLite/16.1.0@0xd34d10cc/testing", # capture
                        "ffmpeg/4.0@0xd34d10cc/testing",               # encoder
                        "spdlog/0.17.0@bincrafters/stable",            # logs
                        "gtest/1.8.1@bincrafters/stable",              # UTs
                        # TODO: move to 0xd34d10cc
                        "prometheus-cpp/0.6.0@d34dpkgs/testing",       # metrics
-                       "jsonformoderncpp/3.5.0@vthiery/stable")       # config
+                       "jsonformoderncpp/3.5.0@vthiery/stable",       # config
```
Comment (pl0q1n) [inline conanfile.py]: Seems like you forgot to add conan remotes to the appveyor and travis files, isn't it?

Comment (0xd34d10cc) [reply conanfile.py]: Which one? All packages are either from conan-center or d34dpkgs.

File: src/app/app.cpp (hunk 1 above)
Comment (0xd34d10cc) [reply src/app/app.cpp]: Because all configuration options are now logged in `Options::read`

File: src/common/options.hpp (hunk 2 above)
Comment (0xd34d10cc) [reply src/common/options.hpp]: `url` is required option, so it does not need default value
default value for `codec` is empty string, which means that codec should be auto-detected
